app-test-demo/src/test/java/com/nortal/test/demo/glue/TodoUiStepDefinitions.java

```diff
@@ -0,0 +1,69 @@
+package com.nortal.test.demo.glue;
+
+import com.codeborne.selenide.CollectionCondition;
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.Selenide;
+import com.nortal.test.demo.hook.FrontendContextualContainer;
+import com.nortal.test.demo.mediator.UiMediator;
+import io.cucumber.java.en.Step;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class TodoUiStepDefinitions {
+    private static final String MAIN_PAGE_HEADER = "//nav/a";
+    private static final String TODO_TABLE_ROWS = "//div/table/tbody/tr";
+    private static final String BUTTON_REMOVE_TODO = "//div/table/tbody/tr[1]/td[4]/button";
+    private static final String BUTTON_ADD_NEW_TODO = "//div/button";
+    public static final String ADD_TODO_PAGE_HEADER = "//div/h1";
+    public static final String DESCRIPTION_INPUT_FIELD = "//div/form/fieldset[1]/input";
+    public static final String SAVE_TODO_BUTTON = "//div/form/button";
+    public static final String DESCRIPTION_VALUE = "test";
+    private final UiMediator uiMediator;
+    private final FrontendContextualContainer frontendContainer;
+    private int initialTodosCount;
+
+    @Step("user has opened Todo application in browser")
+    public void openPageInBrowser() {
+        Selenide.open(frontendContainer.getUiUrl());
+        uiMediator.elementXpath(MAIN_PAGE_HEADER, true).shouldBe(Condition.visible);
```

Comment:
Use page object pattern per this example: 
https://github.com/nordic-institute/X-Road/tree/develop-7.x/src/centralserver/admin-service/ui-system-test/src/intTest/java/org/niis/xroad/cs/test/ui

We should also consider dropping ui mediator as it reduces  code readability

Reply:
created PageObj classes, dropped ui mediator.